packages/utils/src/internals/sitemap.ts

```diff
@@ -98,11 +156,26 @@ export class Sitemap {
 
                 if (sitemapStream.response!.statusCode === 200) {
                     await new Promise((resolve, reject) => {
-                        const parser = Sitemap.createParser(parsingState, () => resolve(undefined), reject);
                         let stream: Duplex = sitemapStream;
                         if (sitemapUrl.endsWith('.gz')) {
                             stream = stream.pipe(createGunzip());
+                            sitemapUrl = sitemapUrl.substring(0, sitemapUrl.length - 3);
                         }
+
+                        const parser = (() => {
+                            const contentType = sitemapStream.response!.headers['content-type'];
+
+                            if (contentType === 'text/xml' || sitemapUrl.endsWith('.xml')) {
+                                return Sitemap.createXmlParser(parsingState, () => resolve(undefined), reject);
+                            }
+
+                            if (contentType === 'text/plain' || sitemapUrl.endsWith('.txt')) {
+                                return new SitemapTxtParser(parsingState, () => resolve(undefined));
```

Comment:
for consistency, I think having a `createTxtParser` method (returning the parser instance) would be nicer

Reply:
I had it that way before, then I found out it just forwarded the arguments to the `SitemapTxtParser` constructor and decided it was useless :shrug: